library/src-3.x/dotty/internal/StringContextMacro.scala

```diff
@@ -0,0 +1,777 @@
+      */
+    def argError(message : String, index : Int) : Unit
+
+    /** Reports error linked with the list of arguments or the StringContext.
+      *
+      * @param message the message to report in the error
+      * @return an error
+      */
+    def strCtxError(message : String) : Unit
+    def argsError(message : String) : Unit
+
+    /** Claims whether an error or a warning has been reported
+      *
+      * @return true if an error/warning has been reported, false
+      */
+    def hasReported() : Boolean
+
+    /** Stores the old value of the reported and reset it to false */
+    def resetReported() : Unit
+
+    /** Restores the value of the reported boolean that has been reset */
+    def restoreReported() : Unit
+  }
+
+  /** Retrieves a String from an Expr containing it
+    *
+    * @param expression the Expr containing the String
+    * @return the String contained in the given Expr
+    * quotes an error if the given Expr does not contain a String
+    */
+  private def literalToString(expression : Expr[String])(implicit reflect: Reflection) : String = expression match {
+    case Const(string : String) => string
+    case _ => QuoteError("Expected statically known part list")
+  }
+
+  /** Retrieves the parts from a StringContext, given inside an Expr, and returns them as a list of Expr of String
+    *
+    * @param strCtxExpr the Expr containing the StringContext
+    * @return a list of Expr containing Strings, each corresponding to one parts of the given StringContext
+    * quotes an error if the given Expr does not correspond to a StringContext
+    */
+  def getPartsExprs(strCtxExpr : Expr[scala.StringContext])(implicit reflect : Reflection): List[Expr[String]] = {
+    import reflect._
+    strCtxExpr.unseal.underlyingArgument match {
+      case Apply(Select(Select(_, "StringContext") | Ident("StringContext"), "apply"), List(parts1)) =>
+        parts1.seal.cast[Seq[String]] match {
+          case ExprSeq(parts2) => parts2.toList
+          case _ => throw new Exception("Expected statically known String Context")
+        }
+      case Apply(Select(New(TypeIdent("StringContext")), _), List(parts1)) =>
+        parts1.seal.cast[Seq[String]] match {
+          case ExprSeq(parts2) => parts2.toList
+          case _ => throw new Exception("Expected statically known String Context")
+        }
+      case _ => QuoteError("Expected statically known String Context") //TODO : quotes?
+    }
+  }
+
+  /** Retrieves a list of Expr, each containing an argument, from an Expr of list of arguments
+    *
+    * @param argsExpr the Expr containing the list of arguments
+    * @return a list of Expr containing arguments
+    * quotes an error if the given Expr does not contain a list of arguments
+    */
+  def getArgsExprs(argsExpr: Expr[Seq[Any]])(implicit reflect: Reflection): List[Expr[Any]] = {
+    import reflect._
+    argsExpr.unseal.underlyingArgument match {
+      case Typed(Repeated(args, _), _) => args.map(_.seal)
```

Comment:
It should probably be
```scala
argsExpr match {
  case ExprSeq(args) => args
  case _ => case tree => QuoteError("Expected statically known argument list", argsExpr)
}
```

Reply:
I tried it and it never matches

Reply:
Leave them like this. I will have a look into it later.